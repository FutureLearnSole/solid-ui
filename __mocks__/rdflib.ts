export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    doc: () => {
      equals: () => {}
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {

}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
}
