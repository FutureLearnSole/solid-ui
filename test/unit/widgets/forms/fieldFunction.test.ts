import { namedNode } from 'rdflib'
import ns from '../../../../src/ns'
import uiStore from '../../../../src/store'

import {
  fieldFunction,
  mostSpecificClassURI
} from '../../../../src/widgets/forms/fieldFunction'

describe('mostSpecificClassURI', () => {
  it('exists', () => {
    expect(mostSpecificClassURI).toBeInstanceOf(Function)
  })
  it('runs', () => {
    const form = namedNode('http://example.com/#form')
    uiStore.add(form, ns.rdf('type'), namedNode('http://example.com/#type'), namedNode('http://example.com/'))
    expect(mostSpecificClassURI(form)).toEqual('http://example.com/#type')
  })
})

describe('fieldFunction', () => {
  it('exists', () => {
    expect(fieldFunction).toBeInstanceOf(Object)
  })
  it('runs', () => {
    expect(fieldFunction(document, namedNode('http://example.com/#this'))).toBeInstanceOf(Function)
  })
})
