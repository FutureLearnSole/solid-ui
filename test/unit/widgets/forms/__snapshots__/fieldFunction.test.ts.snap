// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fieldFunction returns an error block if no matching function exists 1`] = `<div />`;

exports[`fieldFunction returns an error block if subject type undefined 1`] = `
<div
  style="margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: #fee; color:black;"
>
  No handler for field &lt;http://example.com/#doesnt-exist&gt; of type undefined
</div>
`;
